keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyAsyncClient.java

```diff
@@ -0,0 +1,593 @@
+     * key cannot be restored. The key is restored in its entirety with the same key name as it had when it was backed up. If the key name is not
+     * available in the target Key Vault, the RESTORE operation will be rejected. While the key name is retained during restore, the final key identifier
+     * will change if the key is restored to a different vault. Restore will restore all versions and preserve version identifiers. The RESTORE operation is subject
+     * to security constraints: The target Key Vault must be owned by the same Microsoft Azure Subscription as the source Key Vault The user must have RESTORE permission in
+     * the target Key Vault. This operation requires the {@code keys/restore} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Restores the key in the key vault from its backup. Subscribes to the call asynchronously and prints out the restored key
+     * details when the callback happens.</p>
+     * <pre>
+     * //Pass the Key Backup Byte array to the restore operation.
+     * keyAsyncClient.restoreKey(keyBackupByteArray).subscribe(keyResponse -&gt;
+     *   System.out.printf("Restored Key with name %s and id %s \n", keyResponse.value().name(), keyResponse.value().id()));
+     * </pre>
+     *
+     * @param backup The backup blob associated with the key.
+     * @throws ResourceModifiedException when {@code backup} blob is malformed.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link Key restored key}.
+     */
+    public Mono<Response<Key>> restoreKey(byte[] backup) {
+        KeyRestoreRequestParameters parameters = new KeyRestoreRequestParameters().keyBackup(backup);
+        return service.restoreKey(endpoint, API_VERSION, parameters, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+    /**
+     * List keys in the key vault. Retrieves a list of the keys in the Key Vault as JSON Web Key structures that contain the public
+     * part of a stored key. The List operation is applicable to all key types and the individual key response in the flux is represented by {@link KeyBase} as only the base key identifier,
+     * attributes and tags are provided in the response. The key material and individual key versions are not listed in the response. This operation requires the {@code keys/list} permission.
+     *
+     * <p>It is possible to get full keys with key material from this information. Convert the {@link Flux} containing {@link KeyBase base key} to
+     * {@link Flux} containing {@link Key key} using {@link KeyAsyncClient#getKey(KeyBase baseKey)} within {@link Flux#flatMap(Function)}.</p>
+     * <pre>
+     * Flux&lt;Key&gt; keys = keyAsyncClient.listKeys()
+     *   .flatMap(keyAsyncClient::getKey)
+     *   .map(Response::value);
+     * </pre>
+     *
+     * @return A {@link Flux} containing {@link KeyBase key} of all the keys in the vault.
+     */
+    public Flux<KeyBase> listKeys() {
```

Comment:
>Flux<KeyBase> [](start = 11, length = 13)

Shouldn't this be Flux<Response<KeyBase>>?

Reply:
That is not supported in java.